aws/resource_aws_sns_topic.go

```diff
@@ -176,35 +156,223 @@ func resourceAwsSnsTopicCreate(d *schema.ResourceData, meta interface{}) error {
 
 	d.SetId(*output.TopicArn)
 
-	for terraformAttrName, snsAttrName := range SNSAttributeMap {
-		if d.HasChange(terraformAttrName) {
-			_, terraformAttrValue := d.GetChange(terraformAttrName)
-			err := updateAwsSnsTopicAttribute(d.Id(), snsAttrName, terraformAttrValue, snsconn)
-			if err != nil {
-				return err
-			}
+	// update mutable attributes
+	if d.HasChange("application_failure_feedback_role_arn") {
+		_, v := d.GetChange("application_failure_feedback_role_arn")
+		if err := updateAwsSnsTopicAttribute(d.Id(), "ApplicationFailureFeedbackRoleArn", v, snsconn); err != nil {
+			return err
+		}
+	}
+	if d.HasChange("application_success_feedback_role_arn") {
+		_, v := d.GetChange("application_success_feedback_role_arn")
+		if err := updateAwsSnsTopicAttribute(d.Id(), "ApplicationSuccessFeedbackRoleArn", v, snsconn); err != nil {
+			return err
+		}
+	}
+	if d.HasChange("arn") {
```

Comment:
Nit: `arn` is not actually configurable, but presumably it was included in the previous mapping so it would get picked up for read. This is no different than before so it is fine to leave for now. 👍